daemon/api_debug_seeding.go

```diff
@@ -0,0 +1,107 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"time"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+type seedingInfo struct {
+	// true if the device is seeded (same as "seeded" in the state)
+	Seeded bool `json:"seeded,omitempty"`
+
+	// true if snap-preseed was applied
+	Preseeded bool `json:"preseeded,omitempty"`
+
+	// when snap-preseed was started
+	PreseedStartTime time.Time `json:"preseed-start-time,omitempty"`
+
+	// when snap-preseed work finished (i.e. before firstboot)
+	PreseededTime time.Time `json:"preseeded-time,omitempty"`
```

Comment:
PreseedEndTime `preseed-end-time` ? or /End/Complete/

Reply:
the idea was actually to call this "preseed-time" to match "seed-time"